src/controllers/Product.controller.ts

```diff
@@ -81,3 +83,42 @@ export const getAvailableProducts = async (req: Request, res: Response) => {
     }
   }
 };
+
+// Seller update a product
+export const updateProduct = async (req: Request, res: Response) => {
+  try {
+    const productId = req.params.id;
+    const userId = req.user.id;
+    const productToUpdate = await findProductService(productId);
+
+    const parsedProductToUpdate = JSON.parse(JSON.stringify(productToUpdate));
+
+    if (parsedProductToUpdate) {
+      for (const productCriteria in req.body) {
+        if (req.body[productCriteria]) {
+          parsedProductToUpdate[productCriteria] = req.body[productCriteria];
+        }
```

Comment:
Make sure validation happens not within the controller functions, but within the miiddleware.

Reply:
I have done this bro